qutebrowser/browser/downloads.py

```diff
@@ -507,6 +507,25 @@ def __str__(self):
                         remaining=remaining, perc=perc, down=down,
                         total=total, errmsg=errmsg))
 
+    def _find_next_filename(self) -> None:
+        """Append unique numeric suffix to filename.
+
+        After finding a unique filename, restart the process of setting the
+        filename which will update the filename shown in the downloads list."""
+        assert self._filename is not None
+        path = pathlib.Path(self._filename)
+        suffix = ''.join(path.suffixes)
+        base = path.name[:-len(suffix)]
+
+        i = 1
+        while True:
+            i += 1
+            path_to_try = path.parent / '{}_{}{}'.format(base, i, suffix)
```

Comment:
Note you can use f-strings since we dropped Python 3.5 support in qutebrowser 2:

```suggestion
            path_to_try = path.parent / f'{base}_{i}{suffix}'
```